extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1498,13 +1502,18 @@ export class AmpA4A extends AMP.BaseElement {
    * TODO(keithwrightbos@): report error cases
    */
   getAmpAdMetadata_(creative) {
```

Comment:
Also: sorry for not making my overall intentions clearer initially. But basically I wish to simplify / straighten this, not make it more complicated, and I'm willing to suffer temporary complexity but not permanent complexity for it.